fe/fe-core/src/main/java/com/starrocks/load/LockManager.java

```diff
@@ -0,0 +1,249 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.load;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.locks.ReentrantLock;
+
+// non-reentrant lock manager
+public class LockManager {
+    private static final Logger LOG = LogManager.getLogger(LockManager.class);
+    public static enum LockType {
+        LOCK_SHARED,
+        LOCK_EXCLUSIVE
+    };
+
+    public static class PartitionEntity {
+        // if partition id is equal to table id, it means the lock is for table
+        public long id;
+        public volatile long refCount;
+
+        PartitionEntity(long id) {
+            this.id = id;
+            this.refCount = 1;
+        }
+    }
+
+    private ReentrantLock lock;
+    // tableId -> LockType -> partitionIds
+    private Map<Long, Map<LockType, Set<PartitionEntity>>> lockTables;
```

Comment:
why not `Map<partitionId, ReadWriteLock>`
I think it is not a good idea to use LockType as a Map key.

Reply:
I will modify it